// react-native-haptic-feedback/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  trigger: true,
  stop: true,
  isSupported: true,
  triggerPattern: true,
  getSystemHapticStatus: true,
  setEnabled: true,
  isEnabled: true,
  impact: true,
  playAHAP: true,
  pattern: true,
  PATTERN_CHARS: true,
  useHaptics: true,
  Patterns: true,
  playHaptic: true,
  TouchableHaptic: true
};
Object.defineProperty(exports, "PATTERN_CHARS", {
  enumerable: true,
  get: function () {
    return _pattern.PATTERN_CHARS;
  }
});
Object.defineProperty(exports, "Patterns", {
  enumerable: true,
  get: function () {
    return _presets.Patterns;
  }
});
Object.defineProperty(exports, "TouchableHaptic", {
  enumerable: true,
  get: function () {
    return _TouchableHaptic.TouchableHaptic;
  }
});
exports.isSupported = exports.isEnabled = exports.impact = exports.getSystemHapticStatus = exports.default = void 0;
Object.defineProperty(exports, "pattern", {
  enumerable: true,
  get: function () {
    return _pattern.pattern;
  }
});
exports.playAHAP = void 0;
Object.defineProperty(exports, "playHaptic", {
  enumerable: true,
  get: function () {
    return _playHaptic.playHaptic;
  }
});
exports.triggerPattern = exports.trigger = exports.stop = exports.setEnabled = void 0;
Object.defineProperty(exports, "useHaptics", {
  enumerable: true,
  get: function () {
    return _useHaptics.useHaptics;
  }
});
var _hapticFeedback = _interopRequireDefault(require("./hapticFeedback"));
var _pattern = require("./utils/pattern");
var _useHaptics = require("./hooks/useHaptics");
var _presets = require("./presets");
var _types = require("./types");
Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});
var _playHaptic = require("./utils/playHaptic");
var _TouchableHaptic = require("./components/TouchableHaptic");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const {
  trigger,
  stop,
  isSupported,
  triggerPattern,
  getSystemHapticStatus,
  setEnabled,
  isEnabled,
  impact
} = _hapticFeedback.default;
/** @platform ios */
exports.impact = impact;
exports.isEnabled = isEnabled;
exports.setEnabled = setEnabled;
exports.getSystemHapticStatus = getSystemHapticStatus;
exports.triggerPattern = triggerPattern;
exports.isSupported = isSupported;
exports.stop = stop;
exports.trigger = trigger;
const {
  playAHAP
} = _hapticFeedback.default;
exports.playAHAP = playAHAP;
var _default = exports.default = _hapticFeedback.default;
